### ISSUE TITLE
training: train a VLM model to output 3D Bounding box from voxel data

### ISSUE THREAD

**Issue by new5558** [menloresearch/voxel-representation#5]
Using dataset from #4

**Comment by new5558** [comment]
From @bachvudinh https://github.com/menloresearch/voxel-representation/issues/2#issuecomment-2784922578

## Fix the training bug.
- Due to large room dimensions in the data, 2D scans generate high-resolution video sequences that exceed our training capacity. These sequences reach over 12,000 tokens, while our A6000 GPUs can only process sequences up to 4,096 tokens. To address this limitation, I've filtered out these oversized samples and reduced the minimum pixels per unit from 50 to 35 when generating training data.
- The Training loss go down smoothly to 0.23 at the end of epoch:

 ![Image](https://github.com/user-attachments/assets/bdcbef8c-c01d-495e-8fa6-ed043df9fd78)

## Today
- Finish the benchmarking script and Work on the augmentation data with @new5558 .

**Comment by new5558** [comment]
from @bachvudinh  https://github.com/menloresearch/voxel-representation/issues/2#issuecomment-2800343277

## Updated :
- Training run with bounding box not working, likely due to Qwen2.5-vl limitations. We'll pivot to using static images as demonstrated in the VoxRep paper like this:
  
 ![Image](https://github.com/user-attachments/assets/babb1d1b-d195-4877-bbde-2105d5771199)
 
- Also Norapat has resolved the bounding box issue in the data pipeline
### Training Experiment details:

| Run ID | Date       | Model Config | Dataset | Learning Rate | Batch Size | Seq Length | Max Resolution | Loss | Hardware | 
|--------|------------|--------------|---------|---------------|------------|-------|------|----------|----------|
| exp-1 | 2025-04-8 | [Full-Finetune](https://github.com/janhq/visual-thinker/blob/main/training/Llama-factory-config/Qwen2.5_1.5B_distil.yaml) | 6k 2D Room Scan Video | 1e-4 | 32  | 4096 |   300x300 | 0.23 | ~ 1.5 hours on 6xA6000s | 
| exp-2 | 2025-04-9 | [Full-Finetune](https://github.com/janhq/visual-thinker/blob/main/training/Llama-factory-config/Qwen2.5_1.5B_distil.yaml) | 6k 2D Room Scan Video | 1e-4 | 32  | 8192 |   400x400 | 0.2 | ~ 20 hours on 6xA6000s | 
| exp-3 | 2025-04-11 | [Full-Finetune](https://github.com/janhq/visual-thinker/blob/main/training/Llama-factory-config/Qwen2.5_1.5B_distil.yaml) | 6k 2D Room Scan Video without bounding box | 3e-4 | 32  | 8192 |   400x400 | 0.06 | ~ 20 hours on 6xA6000s | 
| exp-4 | 2025-04-11 | [Full-Finetune](https://github.com/janhq/visual-thinker/blob/main/training/Llama-factory-config/Qwen2.5_1.5B_distil.yaml) | Scannet Subset  | 1e-4 | 32  | 8192 |   400x400 |  | ~  on 8xH100 | 
| exp-5 | 2025-04-16 | [Full-Finetune](https://github.com/janhq/visual-thinker/blob/main/training/Llama-factory-config/Qwen2.5_1.5B_distil.yaml) | 26k 2D Room Scan Images without bounding box  | 1e-4 | 64  | 4096 |   1280x1280 | 0.02 | ~14  on 8xA6000 | 
| exp-6 | 2025-04-16 | [Full-Finetune](https://github.com/janhq/visual-thinker/blob/main/training/Llama-factory-config/Qwen2.5_1.5B_distil.yaml) | 26k 2D Room Scan Images without bounding box  | 1e-4 | 64  | 4096 |   896x896 | 0.01 | ~4  on 8xA6000 | 

### Today:
- Start training run with images data with Qwen-2.5Vl and Gemma3-4B[Finished]
